src/Ng/CommandHelpers.cs

```diff
@@ -370,99 +399,78 @@ public static StorageFactory CreateSuffixedStorageFactory(string suffix, IDictio
             {
                 throw new ArgumentNullException("suffix");
             }
-
-            Uri storageBaseAddress = null;
-            string storageBaseAddressStr;
-            if (arguments.TryGetValue("-storageBaseAddress" + suffix, out storageBaseAddressStr))
+            try
             {
-                storageBaseAddressStr = storageBaseAddressStr.TrimEnd('/') + "/";
+                Uri storageBaseAddress = null;
```

Comment:
I've seen this `storageBaseAddress` chunk a couple times, you could probably reduce this into a function you call in all these `StorageFactory` factory methods.